src/lib/json/json.h

```diff
@@ -371,8 +376,6 @@ json_tree_lookup(struct json_tree *tree, struct json_token *parent,
 			ret = parent->children[token->num];
 		break;
 	case JSON_TOKEN_ANY:
-		if (likely(parent->max_child_idx >= 0))
-			ret = parent->children[parent->max_child_idx];
```

Comment:
The fix looks easier than I thought, thanks for the patch! Seems to be a leftover from something, otherwise can't imagine why would it treat `[*]` as last child intentionally. After seeing the fix managed to make even stranger bug (fix by this commit):
```Lua
box.cfg{}
s = box.schema.create_space('test')
s:create_index('pk')
s:create_index('sk1', {parts = {{2, 'unsigned', path = '[1][1]'}}})
s:create_index('sk2', {parts = {{2, 'unsigned', path = '[1][2]'}}})
t = s:replace{1, {{1, 2, 3}}}
t['[2][1][*]'] -- returns 2! It just returns latest indexed field here.
```